### DISCUSSION TITLE
dcm4che-5.19.0.The DICOM file received through storescp is incorrect

### DISCUSSION THREAD

**Discussion by qq1045561395** [dcm4che/dcm4che#1128] [Q&A] [answered]
dcm4che version：5.19.0
environment：Java8

example(5.19.0)：
1、storescp -b PACS:3100 --directory H:/java/dicom
2、movescu -b PACS -c PACSSER@192.168.5.128:4100 -m StudyInstanceUID=1.2.840.113619.186.808617694251.20220207145145090.719 -L STUDY --dest PACS  --dest PACS

Through the above command, the DICOM file size I obtained is only 1.5MB. And it cannot be viewed with radiant, and it prompts no DICOM images found.

example(2.0.29)：
1、dcmqr -L PACS:3100 PACSSER@192.168.5.128:4100 -cmove PACS -qStudyInstanceUID=1.2.840.113619.186.808617694251.20220207145145090.719 -cstore CT -cstore PR:LE -cstoredest H:/java/dicom

When I use dcm4che-2.0.29, the DICOM file obtained through storescp can reach 9mb and can be viewed.
I use the default configuration. I have tried many things, such as modifying transfersyntax and decompressing DICOM in dcm2dcm. A week has passed and the problem has not been solved yet I'm in great pain now. I hope to get help.thank you.

**Comment by gunterze** [comment]
Can the received DICOM file be dumped by dcmdump?

**Reply by qq1045561395** [reply]
dcm4che version:5.19.0
1、dcmdump 1.2.840.113681.2886799122.1644232091.4240.1656

I entered the above command with an error

C:\Users\Administrator\Desktop\dcm4che-5.19.0\bin>dcmdump 1.2.840.113681.2886799122.1644232091.4240.1656
0: [0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\
132: (0002,0000) UL #4 [196] FileMetaInformationGroupLength
144: (0002,0001) OB #2 [0\1] FileMetaInformationVersion
158: (0002,0002) UI #28 [1.2.840.10008.5.1.4.1.1.1.2] MediaStorageSOPClassUID
194: (0002,0003) UI #46 [1.2.840.113681.2886799122.1644232091.4240.1656] Media
248: (0002,0010) UI #20 [1.2.840.10008.1.2.1] TransferSyntaxUID
276: (0002,0012) UI #20 [1.2.840.113619.6.94] ImplementationClassUID
304: (0002,0013) SH #14 [CENTRICITY_4.0] ImplementationVersionName
326: (0002,0016) AE #6 [GEPACS] SourceApplicationEntityTitle
340: (0800,0500) CS #2560 [ISO_IR 10CS DERIVED\PRIMARY  UI ,1.2.840.1
18:09:23,860 WARN  - Unrecognized VR code: 5374H - treat as UN
18:09:23,900 WARN  - IOException during read of (5020,4543) #908082464 @ 2908
java.io.EOFException
        at org.dcm4che3.util.StreamUtils.readFully(StreamUtils.java:74)
        at org.dcm4che3.io.DicomInputStream.readFully(DicomInputStream.java:394)
        at org.dcm4che3.io.DicomInputStream.readValue(DicomInputStream.java:784)
        at org.dcm4che3.tool.dcmdump.DcmDump.readValue(DcmDump.java:123)
        at org.dcm4che3.io.DicomInputStream.readAttributes(DicomInputStream.java:534)
        at org.dcm4che3.io.DicomInputStream.readDataset(DicomInputStream.java:468)
        at org.dcm4che3.tool.dcmdump.DcmDump.parse(DcmDump.java:87)
        at org.dcm4che3.tool.dcmdump.DcmDump.main(DcmDump.java:247)
dcmdump: null

**Reply by qq1045561395** [reply]
I use other programs to get the correct DICOM file, which is like this after dcmdump.

C:\Users\Administrator\Desktop\dcm4che-5.19.0\bin>dcmdump 1.2.840.113681.2886799122.1644232091.4240.1740.2.1
0: [0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\
132: (0002,0000) UL #4 [178] FileMetaInformationGroupLength
144: (0002,0001) OB #2 [0\1] FileMetaInformationVersion
158: (0002,0002) UI #26 [1.2.840.10008.5.1.4.1.1.2] MediaStorageSOPClassUID
192: (0002,0003) UI #50 [1.2.840.113681.2886799122.1644232091.4240.1740.2.1] M
250: (0002,0010) UI #20 [1.2.840.10008.1.2.2] TransferSyntaxUID
278: (0002,0012) UI #16 [1.2.40.0.13.1.1] ImplementationClassUID
302: (0002,0013) SH #12 [dcm4che-2.0] ImplementationVersionName
322: (0008,0005) CS #10 [ISO_IR 100] SpecificCharacterSet
340: (0008,0008) CS #20 [DERIVED\PRIMARY\TOMO] ImageType
368: (0008,0014) UI #44 [1.2.840.113681.2886799122.1608518773.4984.1] Instance
420: (0008,0016) UI #26 [1.2.840.10008.5.1.4.1.1.2] SOPClassUID
454: (0008,0018) UI #50 [1.2.840.113681.2886799122.1644232091.4240.1740.2.1] S
512: (0008,0020) DA #8 [20220207] StudyDate
528: (0008,0021) DA #8 [20220207] SeriesDate
544: (0008,0022) DA #8 [20220207] AcquisitionDate
560: (0008,0023) DA #8 [20220207] ContentDate
576: (0008,0030) TM #6 [152825] StudyTime
590: (0008,0031) TM #6 [153658] SeriesTime
604: (0008,0032) TM #6 [153658] AcquisitionTime
618: (0008,0033) TM #6 [153659] ContentTime
632: (0008,0050) SH #16 [ZH220207HOLO002] AccessionNumber

**Comment by gunterze** [comment, marked as answer]
Would need the log of storescp on Association negotiation and receiving the C-STORE RQ.
It seems that the object was received Big Endian (tsuid: 1.2.840.10008.1.2.2) , but  the prefixed File Meta Information declares it as Little Endian:
```
248: (0002,0010) UI https://github.com/dcm4che/dcm4che/pull/20 [1.2.840.10008.1.2.1] TransferSyntaxUID
```

**Reply by qq1045561395** [reply]
I tried the same DICOM file.Dump it through dcmdump.

version 5.19.0：
The file size is 1639kb.
`[2.16.840.1.113709.1.2.2] TransferSyntaxUID`
`[1.2.840.113619.6.94] ImplementationClassUID`
`[CENTRICITY_4.0] ImplementationVersionName`


version 2.0.29：
The file size is 9591kb.
`[1.2.840.10008.1.2.2] TransferSyntaxUID`
`[1.2.40.0.13.1.1] ImplementationClassUID`
`[dcm4che-2.0] ImplementationVersionName`

Their transfersyntax uid is different.How should I handle it? It seems that storescp has no relevant parameters to set.

**Reply by gunterze** [reply]
```
[2.16.840.1.113709.1.2.2] TransferSyntaxUID
```
seems to be a private Transfer Syntax of GE.
You may restrict the accepted Transfer Syntax by referring a customized [sop-classes.properties](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-assembly/src/etc/storescp/sop-classes.properties) in option
```
    --sop-classes <file|url>         file path or URL of list of accepted
                                     SOP Classes,
                                     resource:sop-classes.properties by
                                     default
```
s.o. https://groups.google.com/g/dcm4che/c/UWT6BNkiF9A

**Reply by qq1045561395** [reply]
I'm in sop-classes.properties, replace all "*" with implicitvrlittleendian

The problem that has bothered me for many days has finally been solved. Thank you very much.Thank dcm4che for your contribution to the open source community. Thank you for your hard work.

**Reply by qq1045561395** [reply]
thank you。da lao six six six

**Reply by kayjet** [reply]
Really appreciated! 
听我说，谢谢你因为有你，温暖了四季。
